src/App/DocumentObject.cpp

```diff
@@ -808,6 +808,74 @@ void DocumentObject::onBeforeChange(const Property* prop)
     signalBeforeChange(*this, *prop);
 }
 
+std::vector<std::pair<Property*, std::unique_ptr<Property>>>
+DocumentObject::onProposedLabelChange(std::string& newLabel)
+{
+    // Note that this work can't be done in onBeforeChangeLabel because FeaturePython overrides this
+    // method and does not initially base-call it.
+
+    // We re only called if the new label differs from the old one, and our code to check duplicates
+    // may not work if this is not the case.
+    std::string oldLabel = Label.getStrValue();
+    assert(newLabel != oldLabel);
+    std::string label;
+    if (!isAttachedToDocument()
+        || (getDocument()->testStatus(App::Document::Restoring)
+            && !getDocument()->testStatus(App::Document::Importing))
+        || getDocument()->isPerformingTransaction()) {
+        return {};
+    }
+    static ParameterGrp::handle _hPGrp;
+    if (!_hPGrp) {
+        _hPGrp = GetApplication().GetUserParameter().GetGroup("BaseApp");
+        _hPGrp = _hPGrp->GetGroup("Preferences")->GetGroup("Document");
+    }

```

Comment:
The change was in the original implementation here: https://github.com/FreeCAD/FreeCAD/pull/17944/files#diff-5ad53cc6f40d2529890be512be95fac89dd0a811b9f2be1c6a057a8dfa2c5ec9L1401

This comment can be marked as resolved

Reply:
As mentioned in my #18124 parameter management is also a source of performance problems, and GetGroup calls are one of the hot spots, so keeping a static of the value is very much worthwhile, since this code runs each time a Label is changed. Although the Group reference can be cached, the actual value lookup must be done every time because it may have been changed by the user.

Once I address #18124 properly it may be less of an issue.